### PR TITLE
Bluetooth: Controller: Kconfig to permit periodic sync event skip

### LINKED ISSUE #49410: Bluetooth: Scan responses with info about periodic adv. sometimes stops being reported
**Describe the bug**
Sometimes after for example a sync termination then no new advertisments with periodic adv. are reported in the `bt_le_scan_cb recv` callback. Restarting scanning will result again with scan responses containing info about periodic adv.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using? - Zephyr SDK and NRF52833
 - Is this a regression? - Yes a regression. No bisect made, but it works with the Zephyr version used in nRF Connect SDK 2.0

**To Reproduce**
Build below application based on the `direction_finding_connectionless_rx` sample (the application does not enable CTE sampling by default but can still reproduce the issue).
[aoa_receiver_multiple.zip](https://github.com/zephyrproject-rtos/zephyr/files/9402234/aoa_receiver_multiple.zip)
Have 2 tags sending periodic advertisements (I used about 50ms interval and about 30 bytes per. adv. data payload, this is probably not needed and won't matter) for example use `direction_finding_connectionless_tx` sample, the `periodic_adv` should also work, but I have not tried.
Then just randomly reset the two tags until the log print `Got peridoic adv. scan rsp` disappears. 

**Expected behavior**
Scan responses keep reporting scan responses with periodic advertisements.

**Impact**
Annoyance for application using periodic advertising sync as restart of scan is needed periodically.

**Logs and console output**
```
** Booting Zephyr OS build zephyr-v3.1.0-3844-g0e560bf8e91e  ***
Starting Connectionless Locator Demo
Bluetooth initialization...success
Scan callbacks register...success.
Periodic Advertising callbacks register...success.
Start scanning...success
Waiting for periodic advertising...
Scan is running...
Got peridoic adv. scan rsp.
(306) Found device sending per adv: 21:D6:DD:B7:69:62 (random)
(306) Creating Periodic Advertising Sync...
success.
Sync index: 0
Got peridoic adv. scan rsp.
Got peridoic adv. scan rsp.
(967) PER_ADV_SYNC[0]: [DEVICE]: 21:D6:DD:B7:69:62 (random) synced, Interval 0x0028 (50 ms), PHY LE 1M
Got peridoic adv. scan rsp.
(1177) Found device sending per adv: 3C:97:D6:E5:A2:40 (random)
(1177) Creating Periodic Advertising Sync...
success.
Sync index: 1
TAGS:0,sync...,-,-,-,-,-,-,-,-,
Got peridoic adv. scan rsp.
(1808) PER_ADV_SYNC[1]: [DEVICE]: 3C:97:D6:E5:A2:40 (random) synced, Interval 0x0028 (50 ms), PHY LE 1M
Got peridoic adv. scan rsp.
TAGS:0,0,-,-,-,-,-,-,-,-,
Got peridoic adv. scan rsp.
Got peridoic adv. scan rsp.
TAGS:0,0,-,-,-,-,-,-,-,-,
Got peridoic adv. scan rsp.
(3343) Found device sending per adv: 0F:46:A7:CE:90:F7 (random)
(3343) Creating Periodic Advertising Sync...
success.
Sync index: 2
Got peridoic adv. scan rsp.
(3784) PER_ADV_SYNC[2]: [DEVICE]: 0F:46:A7:CE:90:F7 (random) synced, Interval 0x0028 (50 ms), PHY LE 1M
TAGS:0,0,0,-,-,-,-,-,-,-,
TAGS:0,0,0,-,-,-,-,-,-,-,
(5460) PER_ADV_SYNC[1]: [DEVICE]: 3C:97:D6:E5:A2:40 (random) sync terminated reason: 31
Device was synced: term_cb: clear device index: 1
(5869) PER_ADV_SYNC[0]: [DEVICE]: 21:D6:DD:B7:69:62 (random) sync terminated reason: 31
Device was synced: term_cb: clear device index: 0
TAGS:-,-,0,-,-,-,-,-,-,-,
Scan is running...
TAGS:-,-,0,-,-,-,-,-,-,-,
TAGS:-,-,0,-,-,-,-,-,-,-,
TAGS:-,-,0,-,-,-,-,-,-,-,
TAGS:-,-,0,-,-,-,-,-,-,-,
(10585) PER_ADV_SYNC[2]: [DEVICE]: 0F:46:A7:CE:90:F7 (random) sync terminated reason: 31
Device was synced: term_cb: clear device index: 2
TAGS:-,-,-,-,-,-,-,-,-,-,
Scan is running...
TAGS:-,-,-,-,-,-,-,-,-,-,
TAGS:-,-,-,-,-,-,-,-,-,-,
TAGS:-,-,-,-,-,-,-,-,-,-,
```
After this no new tags sending per. adv. can be found, note that the `Scan is running...` print is still printed.
The 0 in the `TAGS:0,0,-,-,-,-,-,-,-,-,` print indicate that the tag is synced (if CTE is sampled it shows num CTE per second).

**Environment (please complete the following information):**
 - OS: Windows
 - Toolchain: Zephyr SDK
 - Commit SHA or Version used: 4f84bc8f40311fb7ccd6575790c97b4e7807cc87

### PR DESCRIPTION
**Pull request by cvinayak**
When AUX_ADV_IND or AUX_CHAIN_IND auxiliary channel PDU
scan radio event overlaps Periodic Sync radio event, let
the Periodic Sync radio event be aborted so that the
auxiliary channel PDU can be received to generate Extended
Advertising Reports. Added a new Kconfig to enable this.

Fixes #49410.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c [hunk 1]
```diff
@@ -474,7 +474,8 @@ static int is_abort_cb(void *next, void *curr, lll_prepare_cb_t *resume_cb)
 			struct lll_scan_aux *lll_aux;
 
 			lll_aux = ull_scan_aux_lll_is_valid_get(next);
-			if (!lll_aux) {
+			if (IS_ENABLED(CONFIG_BT_CTLR_SYNC_PERIODIC_SKIP_ON_SCAN_AUX) ||
+			    !lll_aux) {
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c:478]: What is the reason why we do not want this to be the (unconfigurable) default? What is the use case for disabling `BT_CTLR_SYNC_PERIODIC_SKIP_ON_SCAN_AUX`?

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c:478]: Update the Kconfig help with the answer.

Comment (Thalley) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c:478]: Just to make sure I understand what is happening: Enabling this Kconfig will basically prioritize scan AUX events over periodic sync events? 

If that is the case, would it make sense, for a longer term plan, to define a set of priorities for the events, so you can basically do

```c
CONFIG_SYNC_PERIODIC_PRIO=10
CONFIG_SCAN_AUX_PRIO=5
```

or 

```c
CONFIG_SYNC_PERIODIC_PRIO=10
CONFIG_SCAN_AUX_PRIO=15
```

To control the prioritizes? Are there any other events (e.g. ACL connections events or ISO sync receiver events) that may also be impacted?

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c:478]: > Just to make sure I understand what is happening: Enabling this Kconfig will basically prioritize scan AUX events over periodic sync events?

No, I have carefully used the term permit. Zephyr Controller does not use priorities for state/roles, rather the preference is to use round-robin, skip count, age and just-in-time negotiation to determine a fair scheduling result.

I have not yet concluded on a scheduling scheme between auxiliary PDU scan in unreserved time space and other states/roles. For the Periodic Sync event in unreserved time space that is overlapping with other Periodic Sync events, a round-robin scheduling is implemented in https://github.com/zephyrproject-rtos/zephyr/pull/49579. Adding  an age factor would be better, would consider it in the future.

> Are there any other events (e.g. ACL connections events or ISO sync receiver events) that may also be impacted?

A similar implementation will be added for ISO sync in the future. Use of priorities, if not by use of dynamic priorities at runtime will lead to starvation/livelock of certain state/roles.

ACL connections use reserved time space hence use round-robin and being under supervision timeout as the sole conditions of mitigating overlaps.

**Review by Thalley — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**